dashboard/app/models/levels/blockly.rb

```diff
@@ -379,6 +380,33 @@ def localized_blockly_level_options(script)
     options.freeze
   end
 
+  def localize_blockly_level_options_for_lab2(script)
+    options = Rails.cache.fetch("#{cache_key}/#{script.try(:cache_key)}/#{I18n.locale}/localized_blockly_level_options", force: !Unit.should_cache?) do
+      level_options = blockly_level_options.dup
+
+      functions = level_options.
+        try(:[], "levelData").
+        try(:[], "startSources").
+        try(:[], "blocks").try(:[], "blocks")&.
+        filter {|block| block.try(:[], "type") == "procedures_defnoreturn"}
+
+      functions&.each do |function|
+        function_name = function.try(:[], "fields").try(:[], "NAME")
+        next unless function_name
+
+        localized_name = I18n.t(
+          "name",
+          scope: [:data, :function_definitions, name, function_name],
```

Comment:
I may be missing something but i don't think you nee the `name` in the scope.

Reply:
Given the structure we discussed: 
https://github.com/code-dot-org/code-dot-org/pull/61603#discussion_r1797483420
you definitely don't need the name in the scope:
```suggestion
          scope: [:data, :function_definitions, function_name],
```

Reply:
It seems to give a value of `null` when removing `name`.  Also, there are quite a few instances of 
```
scope: [:data, :function_definitions, name,
```
in the same file, so maybe it's needed?

Reply:
Ah! I see. For example, we can see in `behaviour_names` that strings get reorganized by level during the sync out, and `name` is the level's name:
```
"data": {
      "behavior_names": {
        "AllTheThings sprite lab with video": {
          "growing": "creciente",
          "jittering": "nerviosismo",
```